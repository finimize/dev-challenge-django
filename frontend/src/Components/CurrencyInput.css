.currency-input {
	width: 100%;
	margin: auto;
	font-weight: 800;
	white-space: nowrap;
	display: inline-block;
}

.currency-input span {
	background: white;
	display: inline-block;
	font-size: 15px;
	padding: 9px 15px;
	border: 2px solid #AAB8C8;
	border-bottom-left-radius: 8px;
	border-top-left-radius: 8px;
	border-right: none;
}

.currency-input input {
	padding: 9px;
	font-size: 14px;
	display: inline-block;
	border: 2px solid #AAB8C8;
	border-bottom-right-radius: 8px;
	border-top-right-radius: 8px;
	text-align: center;
	letter-spacing: 1px;
	max-width: 150px;
	width: 60%;
}

.currency-input :focus {
	outline: none;
	border: 2px solid #00b2ff;
}

.currency-input .default-value input {
	color: #00b2ff;
}
