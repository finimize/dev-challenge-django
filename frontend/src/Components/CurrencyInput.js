import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './CurrencyInput.css'

export default class CurrencyInput extends Component {
	constructor(props) {
		super(props)

		this.state = {
			hasFocus: false,
			value: props.defaultValue,
			valid: true
		}
	}

	handleChange(e) {
		const name = e.target.name
		const value = e.target.value
		const valid = e.target.validity.valid
		this.setState({value, valid})
		if (valid) {
			this.props.onChange({target: {name, value: parseFloat(value)}})
		}
	}

	handleFocus(e) {
		this.setState({
			hasFocus: true
		})
	}

	render() {
		const { defaultValue, name } = this.props
		const { value } = this.state
		const style = {}
		if (!this.state.valid) {
			style.borderColor = 'red';
		}

		return (
			<div className={`currency-input ${defaultValue !== undefined ? 'default-value' : ''}`}>
				<span>Â£</span>
				<input type="number"
					name={name}
					value={value}
					min="0.00"
					step="0.01"
					style={style}
					onChange={this.handleChange.bind(this)}
					onFocus={this.handleFocus.bind(this)}/>
			</div>
		)
	}
}

CurrencyInput.propTypes = {
	defaultValue: PropTypes.number
}
