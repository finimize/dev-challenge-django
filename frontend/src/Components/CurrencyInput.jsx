import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './CurrencyInput.css';
import isValid from '../helpers';

export default class CurrencyInput extends Component {
  constructor(props) {
    super(props);

    this.state = { value: props.defaultValue };
  }

  handleChange(e) {
    const { value } = e.target;
    const { onChange } = this.props;
    this.setState({ value });
    if (isValid(value)) {
      onChange(parseFloat(value));
    }
  }

  render() {
    const { defaultValue } = this.props;
    const { value } = this.state;

    return (
      <div className={`currency-input ${defaultValue !== undefined ? 'default-value' : ''}`}>
        <span>
Â£
        </span>
        <input
          type="text"
          value={value}
          onChange={(e) => { this.handleChange(e); }}
        />
      </div>
    );
  }
}

CurrencyInput.propTypes = {
  defaultValue: PropTypes.number,
  onChange: PropTypes.func,
};

CurrencyInput.defaultProps = {
  defaultValue: 0,
  onChange: () => {},
};
