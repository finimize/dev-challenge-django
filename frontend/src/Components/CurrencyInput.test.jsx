import React from 'react';
import TestRenderer from 'react-test-renderer';
import CurrencyInput from './CurrencyInput';

describe('CurrencyInput', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(
      <CurrencyInput value={5} onChange={() => {}} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
