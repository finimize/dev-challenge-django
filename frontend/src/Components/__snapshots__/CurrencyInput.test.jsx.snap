// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput renders correctly 1`] = `
<div
  className="currency-input default-value"
>
  <span>
    Â£
  </span>
  <input
    onChange={[Function]}
    type="text"
    value={0}
  />
</div>
`;
